src/rtc/rtc_s1_single_job.py

```diff
@@ -30,6 +30,91 @@
 logger = logging.getLogger('rtc_s1')
 
 
+def compute_correction_lut(burst_in, dem_raster, scratch_path,
+                           rg_step=200,
+                           az_step=0.25):
+    '''
+    Compute lookup table for geolocation correction
+
+    Parameters
+    ----------
+    burst_in: Sentinel1BurstSlc
+        Input burst SLC
+    dem_raster: isce3.io.raster
+        DEM to run rdr2geo
+    scratch_path: str
+        Scratch path where the radargrid rasters will be saved
+    rg_step: float
+        Slant range posting of the radar grid to run rdr2geo. Unit: meters
+    az_step: float
+        grid posting in slant range of the radar grid in meters
+        Azimuth time posting of the radar grid to run rdr2geo. Unit: seconds
+
+    Returns
+    -------
+    rg_lut, az_lut: isce3.core.LUT2d
+        LUT2d for geolocation correction in slant range and azimuth direction
+    '''
+    # Bistatic - azimuth direction
+    bistatic_delay = burst_in.bistatic_delay(range_step=rg_step, az_step=az_step)
+
+    # Calculate rdr2geo rasters
+    epsg = dem_raster.get_epsg()
+    proj = isce3.core.make_projection(epsg)
+    ellipsoid = proj.ellipsoid
+
+    rdr_grid = burst_in.as_isce3_radargrid(az_step=az_step,
+                                        rg_step=rg_step)
+
+    grid_doppler = isce3.core.LUT2d()
+
+    # Initialize the rdr2geo object
+    rdr2geo_obj = isce3.geometry.Rdr2Geo(rdr_grid, burst_in.orbit,
+                                         ellipsoid, grid_doppler,
+                                         threshold=1.0e-8)
+
+    # Get the rdr2geo raster needed for SET computation
+    topo_output = {f'{scratch_path}/height.rdr': gdal.GDT_Float64,
+                   f'{scratch_path}/incidence_angle.rdr': gdal.GDT_Float32}
+    raster_list = [
+        isce3.io.Raster(fname, rdr_grid.width,
+                        rdr_grid.length, 1, dtype, 'ENVI')
+        for fname, dtype in topo_output.items()]
+    height_raster, incidence_raster = raster_list
+
+    rdr2geo_obj.topo(dem_raster, x_raster=None, y_raster=None,
+                     height_raster=height_raster,
+                     incidence_angle_raster=incidence_raster)
```

Comment:
We call `topo` when computing the layover shadow mask. Do you think we can reorganize the code so we call `topo` only once?